### PR TITLE
Add an option to fail the test if it times out and enable on Travis

### PR DESCRIPTION
**Pull request by ylemkimon**
* Added `fail_on_timeout` option to fail the test by raising TimeOutError and print traceback if it times out.
* Enabled 595 seconds timeout on Travis builds. ([Example](https://travis-ci.org/ylemkimon/sympy/jobs/280894080))

See comments of #13355.

### CODE REVIEW

File: sympy/utilities/runtests.py [hunk 1]
```diff
@@ -1059,7 +1067,8 @@ def __init__(self, reporter, kw="", post_mortem=False,
         else:
             self._slow_threshold = 10
 
-    def test(self, sort=False, timeout=False, slow=False, enhance_asserts=False):
+    def test(self, sort=False, timeout=False, fail_on_timeout=False, slow=False,
+            enhance_asserts=False):
```
Comment (smichr) [inline sympy/utilities/runtests.py]: To preserve arguments passed positionally, this new parameter should be added to the end of the parameter list, shouldn't it?

File: sympy/utilities/runtests.py [hunk 2]
```diff
@@ -1113,7 +1123,8 @@ def visit_Assert(self, stmt):
         new_tree = Transform().visit(tree)
         return fix_missing_locations(new_tree)
 
-    def test_file(self, filename, sort=True, timeout=False, slow=False, enhance_asserts=False):
+    def test_file(self, filename, sort=True, timeout=False, fail_on_timeout=False,
```
Comment (smichr) [inline sympy/utilities/runtests.py]: ditto regarding parameter order

Comment (ylemkimon) [reply sympy/utilities/runtests.py]: @smichr Fixed.

**Comment by smichr** [comment]
Unless the test that is timing out passes we won't be able to commit this.

**Pull request merged**